be/src/storage/rowset/column_iterator_wrapper.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "common/macros.h"
+#include "common/ownership.h"
+#include "storage/rowset/column_iterator.h"
+
+namespace starrocks {
+
+class ColumnIteratorWrapper : public ColumnIterator {
+public:
+    explicit ColumnIteratorWrapper(ColumnIterator* iter, Ownership ownership) : _iter(iter), _ownership(ownership) {}
```

Comment:
provide a `parent()` interface for accessing its original ColumnIterator directly.

Reply:
Updated